src/RedisClusterOptionsFromIni.php

```diff
@@ -88,16 +88,16 @@ public function __construct()
             $authenticationConfiguration = '';
         }
 
-        $authenticationByName = [];
+        $passwordByName = [];
```

Comment:
So the username cannot be provided via ini configuration?
Do I get this right?

Reply:
Well technically I think it can be provided as follows:
`     
     redis.clusters.seeds = "mycluster[]=localhost:7000&test[]=localhost:7001"
     redis.clusters.timeout = "mycluster=5"
     redis.clusters.read_timeout = "mycluster=10"
     redis.clusters.auth = "mycluster=password" OR ['user' => 'foo', 'pass' => 'bar] 
 `
 but since I am not so sure on the format and RedisClusterOptionsFromIni is designed for getting ini variables by the clustername (which is not provided if you use a username), I did not want to implement this in this PR.
 
 If there really will be a need for this in the future, I would like to honestly see this as a separate PR

Reply:
Yah, I do not think that `redis.cluster.auth` can be formatted like this. Internally, phpredis is handling that as `string`, so there is no way to have it formatted as an array. You can but that won't work. 

I had a quick glance on the C code. Pretty sure that the value will simply persisted as password via https://github.com/phpredis/phpredis/blob/4cd3f59356582a65aec1cceed44741bd5d161d9e/library.c#L4396

So unless they change anything there, I guess we can assume that via that `auth` ini setting, only passwords can be provided. If they continue with the existing logic (using URI format), they might introduce something like `<username>:<password>`. But this would require a new major along with users changing the way how they store the password right now (since passwords containing colon `:` will need to be migrated to URI value `%3A`) - I do not expect that to happen and thus, I rather expect some new setting such as `redis.clusters.users` (maybe they also rename `redis.clusterss.auth` to `redis.clusters.passwords` at some point) - but I'd say lets wait until then.